core/controllers/admin_test.py

```diff
@@ -964,6 +964,179 @@ def test_changing_user_role_from_exploration_editor_to_topic_manager(self):
+                        'height': '1d525',
+                        'width': '3d33',
+                        'verticalPadding': '0d241'
+                    }
+                }
+            }
+        }
+        csrf_token = self.get_new_csrf_token()
+        svg_file_1 = (
+            '<svg xmlns="http://www.w3.org/2000/svg" width="1.33ex" height="1.4'
+            '29ex" viewBox="0 -511.5 572.5 615.4" focusable="false" style="vert'
+            'ical-align: -0.241ex;"><g stroke="currentColor" fill="currentColo'
+            'r" stroke-width="0" transform="matrix(1 0 0 -1 0 0)"><path stroke'
+            '-width="1" d="M52 289Q59 331 106 386T222 442Q257 442 2864Q412 404'
+            ' 406 402Q368 386 350 336Q290 115 290 78Q290 50 306 38T341 26Q37'
+            '8 26 414 59T463 140Q466 150 469 151T485 153H489Q504 153 504 145284'
+            ' 52 289Z"/></g></svg>'
+        )
+        svg_file_2 = (
+            '<svg xmlns="http://www.w3.org/2000/svg" width="3.33ex" height="1.5'
+            '25ex" viewBox="0 -511.5 572.5 615.4" focusable="false" style="vert'
+            'ical-align: -0.241ex;"><g stroke="currentColor" fill="currentColo'
+            'r" stroke-width="0" transform="matrix(1 0 0 -1 0 0)"><path stroke'
+            '-width="1" d="M52 289Q59 331 106 386T222 442Q257 442 2864Q412 404'
+            ' 406 402Q368 386 350 336Q290 115 290 78Q290 50 306 38T341 26Q37'
+            '8 26 414 59T463 140Q466 150 469 151T485 153H489Q504 153 504 145284'
+            ' 52 289Z"/></g></svg>'
+        )
+
+        exploration1 = exp_domain.Exploration.create_default_exploration(
+            'exp_id1', title='title1', category='category')
+
+        exp_services.save_new_exploration(editor_id, exploration1)
+        exp_models.ExplorationMathRichTextInfoModel(
+            id='exp_id1',
+            math_images_generation_required=True,
+            latex_strings_without_svg=['+,+,+,+', '\\frac{x}{y}'],
+            estimated_max_size_of_images_in_bytes=20000).put()
+
+        response_dict = self.post_json(
+            feconf.ADMIN_MATH_SVG_IMAGE_GENERATION_HANDLER,
+            {'latexMapping': post_data},
+            csrf_token=csrf_token,
+            upload_files=(
+                ('latex_id1', 'latex_id1', svg_file_1),
+                ('latex_id2', 'latex_id2', svg_file_2), ),
+            expected_status_int=200)
+        self.assertEqual(
+            response_dict,
+            {'number_of_explorations_updated': '1'})
+        self.logout()
+
+    def test_post_svgs_when_some_images_are_not_supplied(self):
+        user_email = 'user1@example.com'
+        username = 'user1'
+
+        self.signup(user_email, username)
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        editor_id = self.get_user_id_from_email(user_email)
+
+        post_data = {
+            'exp_id1': {
+                '+,+,+,+': {
+                    'latexId': 'latex_id1',
+                    'dimensions': {
```

Comment:
You should probably create a domain object for "dimensions" and for any nested dicts that have fixed keys. This structure is quite complex. (See other comments below too.)

Reply:
I've used the domain objects at other places, but this is different because this dict represents the data which is sent to the controller from the frontend.